azurerm/resource_arm_batch_pool.go

```diff
@@ -312,6 +352,13 @@ func resourceArmBatchPoolCreate(d *schema.ResourceData, meta interface{}) error
 		},
 	}
 
+	certificates := d.Get("certificate").([]interface{})
+	certificateReferences, err := azure.ExpandBatchPoolCertificateReferences(certificates)
+	if err != nil {
+		return fmt.Errorf("Error creating Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
```

Comment:
this is an error expanding the `certificate` block, as such we probably want this to be:

```suggestion
		return fmt.Errorf("Error expanding `certificate`: %+v", err)
```